disco_aws_automation/disco_autoscale.py

```diff
@@ -210,35 +205,65 @@ def create_group(self, hostclass, launch_config, vpc_zone_id,
         throttled_call(self.connection.create_auto_scaling_group, group)
         return group
 
+    # pylint: disable=too-many-arguments
     def get_group(self, hostclass, launch_config, vpc_zone_id=None,
                   min_size=None, max_size=None, desired_size=None,
                   termination_policies=None, tags=None,
-                  load_balancers=None):
+                  load_balancers=None, create_if_exists=False,
+                  group_name=None):
         '''
         Returns autoscaling group.
         This updates an existing autoscaling group if it exists,
         otherwise this creates a new autoscaling group.
 
         NOTE: Deleting tags is not currently supported.
-        NOTE: Detaching ELB is not currently supported.
         '''
-        group = self.get_existing_group(hostclass)
-        if group:
-            return self.update_group(
-                group=group, launch_config=launch_config, vpc_zone_id=vpc_zone_id,
-                min_size=min_size, max_size=max_size, desired_size=desired_size,
-                termination_policies=termination_policies, tags=tags, load_balancers=load_balancers)
-        else:
+        # Check if an autoscaling group already exists.
+        group = self.get_existing_group(hostclass=hostclass, group_name=group_name)
+        if create_if_exists or not group:
             return self.create_group(
                 hostclass=hostclass, launch_config=launch_config, vpc_zone_id=vpc_zone_id,
                 min_size=min_size, max_size=max_size, desired_size=desired_size,
                 termination_policies=termination_policies, tags=tags, load_balancers=load_balancers)
+        else:
+            return self.update_group(
+                group=group, launch_config=launch_config,
+                vpc_zone_id=vpc_zone_id, min_size=min_size, max_size=max_size, desired_size=desired_size,
+                termination_policies=termination_policies, tags=tags, load_balancers=load_balancers)
+
+    def get_existing_groups(self, hostclass=None, group_name=None):
+        """
+        Returns all autoscaling groups for a given hostclass, sorted by most recent creation. If no
+        autoscaling groups can be found, returns an empty list.
+        """
+        groups = list(self._get_group_generator())
+        filtered_groups = []
+        for group in groups:
+            filters = [
+                not hostclass or self.get_hostclass(group.name) == hostclass,
+                not group_name or group.name == group_name]
+            if all(filters):
+                filtered_groups.append(group)
```

Comment:
I ended up tossing this for a list comprehension because a suggestion was made to push the group_name filtering lower into the generator method, and a list of one filter just seems like a bit much.